python/ml4ir/applications/ranking/features/feature_fns/string.py

```diff
@@ -0,0 +1,65 @@
+import tensorflow as tf
+
+from ml4ir.base.features.feature_fns.base import BaseFeatureLayerOp
+from ml4ir.base.io.file_io import FileIO
+
+
+class QueryLength(BaseFeatureLayerOp):
+    """
+    Compute the length of the query string context feature
+    """
+    LAYER_NAME = "query_len"
+
+    TOKENIZE = "tokenize"
+    SEPARATOR = "sep"
+
+    def __init__(self, feature_info: dict, file_io: FileIO, **kwargs):
+        """
+        Initialize layer to define a theoretical min max normalization
+
+        Parameters
+        ----------
+        feature_info : dict
+            Dictionary representing the configuration parameters for the specific feature from the FeatureConfig
+        file_io : FileIO object
+            FileIO handler object for reading and writing
+
+        Notes
+        -----
+        Args under feature_layer_info:
+            tokenize: boolean
```

Comment:
we probably want to provide this function an arbitrary tokenizer, in case we want to add support for other languages later.

Reply:
That is true. I'll add a note.